snakebite/client.py

```diff
@@ -393,6 +394,60 @@ def _handle_rename(self, path, node, dst):
         response = self.service.rename(request)
         return {"path": path, "result": response.result}
 
+
+    def rename_dont_move(self, path, dst, overwriteDest=False):
+        ''' Rename (but don't move) path to a destination
+
+        Since you can't move with this operation, and only rename, it would
+        not make sense to pass multiple paths to rename to a single
+        destination. This method uses the underlying rename2 method.
+
+        https://github.com/apache/hadoop/blob/ae91b13/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/ClientProtocol.java#L483-L523
+
+        This method only wraps the FileAlreadyExistsException exception
+        (both for the Java FileAlreadyExistsException error and for "rename
+        destination directory is not empty" when overwriteDest=True). The other
+        exceptions will just be passed along.
+
+        :param path: Source path
+        :type path: string
+        :param dst: destination
+        :type dst: string
+        :returns: A dictionary or None
+        '''
+        if not path:
+            raise InvalidInputException("rename_dont_move: no path given")
+        if not dst:
+            raise InvalidInputException("rename_dont_move: no destination given")
+
+        processor = lambda path, node, dst=dst, overwriteDest=overwriteDest: self._handle_rename2(path, node, dst, overwriteDest)
+        for item in self._find_items([path], processor, include_toplevel=True):
+            return item
+
+
+    def _handle_rename2(self, path, node, dst, overwriteDest):
+        if not dst.startswith("/"):
+            dst = self._join_user_path(dst)
+
+        # Strip the last / if there is one. Hadoop doesn't like this
+        if dst.endswith("/"):
+            dst = dst[:-1]
+
+        request = client_proto.Rename2RequestProto()
+        request.src = path
+        request.dst = dst
+        request.overwriteDest = overwriteDest
+        try:
+            self.service.rename2(request)
+        except RequestError, ex:
+            if ("FileAlreadyExistsException" in str(ex) or
+                "rename destination directory is not empty" in str(ex)):
+                raise FileAlreadyExistsException(ex)
+            else:
+                raise
+
+        return {"path": path}
```

Comment:
Because `rename2`, unlike `rename`, has no return value. I will however mention this in the docs, as I see that's it's confusing.